addons/website_event_booth_sale/models/sale_order.py

```diff
@@ -23,33 +23,47 @@ def _cart_find_product_line(self, product_id=None, line_id=None, **kwargs):
         return lines
 
     def _website_product_id_change(self, order_id, product_id, qty=0, **kwargs):
+        if self._context.get('pricelist') != self.pricelist_id.id:
+            self = self.with_context(pricelist=self.pricelist_id.id)
+
         values = super(SaleOrder, self)._website_product_id_change(order_id, product_id, qty=qty, **kwargs)
         event_booth_pending_ids = kwargs.get('event_booth_pending_ids')
         if event_booth_pending_ids:
             order_line = self.env['sale.order.line'].sudo().search([
                 ('id', 'in', self.order_line.ids),
                 ('event_booth_pending_ids', 'in', event_booth_pending_ids)])
             booths = self.env['event.booth'].browse(event_booth_pending_ids)
-            new_registration_ids = [Command.create({
-                                        'event_booth_id': booth.id,
-                                        **kwargs.get('registration_values'),
-                                    }) for booth in booths]
-            if order_line:
-                event_booth_registration_ids = [Command.delete(reg.id)
-                                                for reg in order_line.event_booth_registration_ids] + new_registration_ids
-            else:
-                event_booth_registration_ids = new_registration_ids
+            if order_line.event_booth_pending_ids.ids != event_booth_pending_ids:
```

Comment:
This change looks a bit unrelated to the price display.
Can you explain?

Would we create registrations too many times before?

Reply:
Indeed it's not related to the price issue !
I just consider useless to delete and recreate the registrations each time the method is called